src/allmydata/storage/http_server.py

```diff
@@ -878,22 +907,14 @@ def listen_tls(
     """
     endpoint = _TLSEndpointWrapper.from_paths(endpoint, private_key_path, cert_path)
 
-    def build_nurl(listening_port: IListeningPort) -> DecodedURL:
-        nurl = DecodedURL().replace(
-            fragment="v=1",  # how we know this NURL is HTTP-based (i.e. not Foolscap)
-            host=hostname,
-            port=listening_port.getHost().port,
-            path=(str(server._swissnum, "ascii"),),
-            userinfo=(
-                str(
-                    get_spki_hash(load_pem_x509_certificate(cert_path.getContent())),
-                    "ascii",
-                ),
-            ),
-            scheme="pb",
+    def get_nurl(listening_port: IListeningPort) -> DecodedURL:
+        return build_nurl(
+            hostname,
+            listening_port.getHost().port,
```

Comment:
It's not a new issue here, it just shows up in this refactoring, but I guess we have no guarantee at this level that the listening port is listening on IPv4/IPv6 so this `getHost` might not return something with a `port`.

I expect that because of what other levels do, we _do_ always have IPv4 or IPv6 here though.  And I'm not sure if it would ever be very interesting to have any other address family anyway.

This is not actionable, I guess.